server/qrops.go

```diff
@@ -0,0 +1,291 @@
+package server
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"image"
+	"net/url"
+	"strconv"
+
+	"github.com/yeqown/go-qrcode/v2"
+	"github.com/yeqown/go-qrcode/writer/standard"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/multiaccounts"
+)
+
+type WriterCloserByteBuffer struct {
+	*bytes.Buffer
+}
+
+func (wc WriterCloserByteBuffer) Close() error {
+	return nil
+}
+
+func NewWriterCloserByteBuffer() *WriterCloserByteBuffer {
+	return &WriterCloserByteBuffer{bytes.NewBuffer([]byte{})}
+}
+
+type QRConfig struct {
+	DecodedQRURL    string
+	WithLogo        bool
+	CorrectionLevel qrcode.EncodeOption
+	KeyUID          string
+	ImageName       string
+	Size            int
+	Params          url.Values
+}
+
+func NewQRConfig(params url.Values, logger *zap.Logger) *QRConfig {
+	config := &QRConfig{}
+	config.Params = params
+	err := config.setQrURL()
+
+	if err != nil {
+		logger.Error("[qrops-error] error in setting QRURL", zap.Error(err))
+	}
```

Comment:
If there is a fail on setting the QR URL, should subsequent processes execute? Or should the process terminate at this point?

Reply:
Its better to terminate here itself because if there is a fail in setting URL other parts of the code would also fail and it would be cumbersome to go through the stack trace to figure out root cause. 
Good catch @Samyoul !

Reply:
fixed here : https://github.com/status-im/status-go/pull/3172/commits/8d1a34bcda651166d09ea2672711e0b662200d54